File: registration_system_rs/src/notification/domain/notification.rs
use chrono::NaiveDateTime;

#[derive(Debug, Clone)]
pub struct Notification {
    pub id: String,
    pub user_id: i64,
    pub kind: String,
    pub title: String,
    pub content: String,
    pub related_type: Option<String>,
    pub related_id: Option<String>,
    pub read_at: Option<NaiveDateTime>,
    pub created_at: NaiveDateTime,
    pub updated_at: NaiveDateTime,
}

File: registration_system_go/internal/notification/domain/notification.go
package domain

import (
	"strings"
	"time"

	sharederror "github.com/oryjk/registration_system/registration_system_go/internal/shared/domain"
)

// Notification 站内通知；kind 由业务方定义（如 teamfund_depleted）。
type Notification struct {
	ID          int64
	UserID      int64
	Kind        string
	Title       string
	Content     string
	RelatedType string
	RelatedID   string
	ReadAt      *time.Time
	CreatedAt   time.Time
}

func (n Notification) IsRead() bool { return n.ReadAt != nil }

func NewNotification(userID int64, kind, title, content, relatedType, relatedID string, now time.Time) (Notification, error) {
	kind = strings.TrimSpace(kind)
	title = strings.TrimSpace(title)
	content = strings.TrimSpace(content)
	if userID <= 0 || kind == "" || title == "" || content == "" {
		return Notification{}, sharederror.New(sharederror.KindValidation, "通知参数无效")
	}
	return Notification{
		UserID: userID, Kind: kind, Title: title, Content: content,
		RelatedType: strings.TrimSpace(relatedType), RelatedID: strings.TrimSpace(relatedID),
		CreatedAt: now,
	}, nil
}
